src/soundanalysis.cs

```diff
@@ -0,0 +1,157 @@
+		[Export ("initWithFormat:")]
+		IntPtr Constructor (AVAudioFormat format);
+
+		[Export ("addRequest:withObserver:error:")]
+		bool AddRequest (ISNRequest request, ISNResultsObserving observer, [NullAllowed] out NSError error);
+
+		[Export ("removeRequest:")]
+		void RemoveRequest (ISNRequest request);
+
+		[Export ("removeAllRequests")]
+		void RemoveAllRequests ();
+
+		[Export ("analyzeAudioBuffer:atAudioFramePosition:")]
+		void Analyze (AVAudioBuffer audioBuffer, long audioFramePosition);
+
+		[Export ("completeAnalysis")]
+		void CompleteAnalysis ();
+	}
+
+	delegate void SNAudioFileAnalyzerAnalyzeHandler (bool didReachEndOfFile);
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNAudioFileAnalyzer {
+
+		[DesignatedInitializer]
+		[Export ("initWithURL:error:")]
+		IntPtr Constructor (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("addRequest:withObserver:error:")]
+		bool AddRequest (ISNRequest request, ISNResultsObserving observer, [NullAllowed] out NSError error);
+
+		[Export ("removeRequest:")]
+		void RemoveRequest (ISNRequest request);
+
+		[Export ("removeAllRequests")]
+		void RemoveAllRequests ();
+
+		[Export ("analyze")]
+		void Analyze ();
+
+		[Async]
+		[Export ("analyzeWithCompletionHandler:")]
+		void Analyze (SNAudioFileAnalyzerAnalyzeHandler completionHandler);
+
+		[Export ("cancelAnalysis")]
+		void CancelAnalysis ();
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNClassification {
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("confidence")]
+		double Confidence { get; }
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNClassificationResult : SNResult {
+
+		[Export ("classifications")]
+		SNClassification [] Classifications { get; }
+#if !WATCH // Remove when CoreMedia is enabled on watchOS
+		[Export ("timeRange")]
+		CMTimeRange TimeRange { get; }
+#endif
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNClassifySoundRequest : SNRequest {
+
+		[Export ("overlapFactor")]
+		double OverlapFactor { get; set; }
+
+		[return: NullAllowed]
+		[Export ("initWithMLModel:error:")]
+		IntPtr Constructor (MLModel mlModel, [NullAllowed] out NSError error);
+	}
+
+	interface ISNRequest { }
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[Protocol]
+	interface SNRequest {
+
+	}
+
+	interface ISNResult { }
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[Protocol]
+	interface SNResult {
+
+	}
+
+	interface ISNResultsObserving { }
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[Protocol]
+	interface SNResultsObserving {
```

Comment:
We could remove the Request prefix, so DidProduceResult, DodFail (No need the with error) and DidComplete.